### PR TITLE
API - Post RealEstate Validations

### PR DESCRIPTION
**Pull request by diogo-s4ntos**
Added validations to method POST of RealEstate.

### CODE REVIEW

**Review by jonathan-osorio — CHANGES_REQUESTED** [review]
The validations are returning more than one message to the same rule: Ex.:
  "Price": [
      "'Price' deve ser definido.",
      "A valid price is required."
    ]
    
    Also there is no cascading on the rules. Maybe we should add it as validator oprions in the Program.cs so the rules don't overlap in the same fields, for exemple the not null and the lenght rules.

File: RealestateApi/Controllers/RealEstateController.cs [hunk 1]
```diff
@@ -52,6 +55,13 @@ public async Task<ActionResult<List<RealEstateRequestDto>>> GetAllRealEstate()
         [HttpPost(Name = "AddRealEstate")]
         public async Task<ActionResult<RealEstateModel>> AddRealEstateAsync(AddRealEstateRequestDto realEstateDto)
         {
+            var validationResult = _realEstateRequestValidatorDto.Validate(realEstateDto);
+
+            if (!validationResult.IsValid)
```
Comment (Silva096) [inline RealestateApi/Controllers/RealEstateController.cs:60]: I suspect that if you remove this If statement, and line 58, the validator will be triggered automatically.
Can you give it a try and let me know please?

File: RealestateApi/Validators/AddRealEstateValidator.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        /// <summary>
+        /// 
+        /// Validate if reference data id exists in RefData
+        /// 
+        /// </summary>
+        /// <param name="refDataType"> Reference Data Type </param>
+        /// <param name="refDataId"> Reference Data Id </param>
+        /// <returns> True/false if reference data exists or not in RefData </returns>
+        public bool ExistInRefData(string refDataType, int refDataId)
```
Comment (Silva096) [inline RealestateApi/Validators/AddRealEstateValidator.cs:65]: This can be changed to private instead of public

**Review by pmzpro — CHANGES_REQUESTED** [review]
There's an error if you insert an inexisting id for customerId and agentId:

The api return the error 500 but should return the error 400 instead.

**Review by pmzpro — APPROVED** [review]

File: RealestateApi/Controllers/RealEstateController.cs [hunk 3]
```diff
@@ -13,7 +14,7 @@ public class RealEstateController : ControllerBase
         private readonly ILogger<RealEstateController> _logger;
         private readonly IRealEstateService _realEstateService;
 
-        public RealEstateController(ILogger<RealEstateController> logger, IRealEstateService realEstateService)
+        public RealEstateController(ILogger<RealEstateController> logger, IRealEstateService realEstateService, IValidator<AddRealEstateRequestDto> realEstateRequestValidatorDto)
```
Comment (Silva096) [inline RealestateApi/Controllers/RealEstateController.cs:17]: remove validator from constructor.

**Review by Silva096 — APPROVED** [review]

**Pull request merged**